packages/core/src/Engine.ts

```diff
@@ -309,32 +309,39 @@ export class Engine extends EventDispatcher {
     this._spriteMaskRenderDataPool.resetPool();
     this._textRenderDataPool.resetPool();
 
+    const { inputManager, _physicsInitialized: physicsInitialized, _waitingDestroy: waitingDestroy } = this;
+    inputManager._update();
+
     const loopScenes = this._sceneManager._scenes.getLoopArray();
-    for (let i = 0, n = loopScenes.length; i < n; i++) {
+    const n = loopScenes.length;
+    for (let i = 0; i < n; i++) {
       const scene = loopScenes[i];
       if (scene.destroyed) {
         continue;
       }
       scene._cameraNeedSorting && scene._sortCameras();
+      scene._componentsManager.callScriptOnStart();
+      physicsInitialized && scene.physics._update(deltaTime);
+    }
 
-      const componentsManager = scene._componentsManager;
-      componentsManager.callScriptOnStart();
-
-      this._physicsInitialized && scene.physics._update(deltaTime);
-
-      this.inputManager._update();
+    physicsInitialized && inputManager._updateByScene(loopScenes);
```

Comment:
`_updateByScene` function name is not clear!
Maybe `_firePointScript`

Reply:
Done.